lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +131,31 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _add_instances_to_groups(self, use_public_ip):
         """Add instance names to their dynamic inventory groups."""
         for instance in self.instances:
-            self.inventory.add_host(instance.label, group=instance.group)
+            if use_public_ip:
+                dest = self._get_instance_public_ip(instance)
```

Comment:
Can we name `dest` better? I feel like `host_identifier` (or so) might be more readable.

Reply:
Will change